docs/common-sdk-requirements.md

```diff
@@ -0,0 +1,43 @@
+# Requirements for the Common SDK infrastructure
+The purpose of this doc is to have a place to list and discuss all requirements to the new modular SDK infrastructure (also informally called "the Common SDK infrastructure") before creating an official design.
+
+The information listed here was primarily sourced in a few meetings/discussions/whiteboardings we had before.
+
+## Terminology
+* _Common SDK_ is the new modular SDK. 
+* _Service_ is a single building unit (module) of the new modular SDK. TODO: please let me know if another term is desired for this.
+
+## Requirements
```

Comment:
The SDK core should use the least common denominator for the specific platform and minimize dependencies on additional packages/libraries. In C# land, this might mean we use .Net Core / Mono runtime only while in C++ we should avoid STL data passing in favor of flat C APIs for cross compatibility

Reply:
ABSOLUTELY agree. This is a SUPER important detail, and I think you're right to describe it explicitly. (this document should as well)